Source/Core/Common/MemArenaDarwin.cpp

```diff
@@ -0,0 +1,225 @@
+// Copyright 2025 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Common/MemArena.h"
+
+#include "Common/Assert.h"
+#include "Common/Logging/Log.h"
+
+namespace Common
+{
+MemArena::MemArena() = default;
+MemArena::~MemArena() = default;
+
+void MemArena::GrabSHMSegment(size_t size, std::string_view base_name)
+{
+  kern_return_t retval = vm_allocate(mach_task_self(), &m_shm_address, size, VM_FLAGS_ANYWHERE);
+  if (retval != KERN_SUCCESS)
+  {
+    ERROR_LOG_FMT(MEMMAP, "GrabSHMSegment failed: vm_allocate returned {0:#x}", retval);
+
+    m_shm_address = 0;
+
+    return;
+  }
+
+  memory_object_size_t entry_size = size;
+  const vm_prot_t prot = VM_PROT_READ | VM_PROT_WRITE;
+
+  retval = mach_make_memory_entry_64(mach_task_self(), &entry_size, m_shm_address, prot,
+                                     &m_shm_entry, MACH_PORT_NULL);
+  if (retval != KERN_SUCCESS)
+  {
+    ERROR_LOG_FMT(MEMMAP, "GrabSHMSegment failed: mach_make_memory_entry_64 returned {0:#x}",
+                  retval);
+
+    m_shm_address = 0;
+    m_shm_entry = MACH_PORT_NULL;
+
+    return;
+  }
+
+  m_shm_size = size;
+}
+
+void MemArena::ReleaseSHMSegment()
+{
+  if (m_shm_entry != MACH_PORT_NULL)
+  {
+    mach_port_deallocate(mach_task_self(), m_shm_entry);
+  }
+
+  if (m_shm_address != 0)
+  {
+    vm_deallocate(mach_task_self(), m_shm_address, m_shm_size);
+  }
+
+  m_shm_address = 0;
+  m_shm_size = 0;
+  m_shm_entry = MACH_PORT_NULL;
+}
+
+void* MemArena::CreateView(s64 offset, size_t size)
+{
+  if (m_shm_address == 0)
+  {
+    ERROR_LOG_FMT(MEMMAP, "CreateView failed: no shared memory segment allocated");
+    return nullptr;
+  }
+
+  vm_address_t address;
+  vm_prot_t prot = VM_PROT_READ | VM_PROT_WRITE;
```

Comment:
Same here for `constexpr` (or at least `const` if that doesn't work for whatever reason), since this is only an input parameter.